omod-2.2/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_2/ObsController2_2Test.java

```diff
@@ -0,0 +1,84 @@
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.webservices.rest.web.v1_0.controller.openmrs2_2;
+
+import org.junit.Test;
+import org.openmrs.Obs;
+import org.openmrs.api.ObsService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.web.RestTestConstants1_8;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public class ObsController2_2Test extends MainResourceControllerTest {
+
+    private static final DateFormat DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS");
+
+    /**
+     * @see MainResourceControllerTest#getURI()
+     */
+    @Override
+    public String getURI() {
+        return "obs";
+    }
+
+    @Override
+    public long getAllCount() {
+        return Context.getObsService().getObservationCount(null, true);
+    }
+
+    /**
+     * @see MainResourceControllerTest#getUuid()
+     */
+    @Override
+    public String getUuid() {
+        return RestTestConstants1_8.OBS_UUID;
+    }
+
+    /**
+     * @see MainResourceControllerTest#shouldGetAll()
+     */
+    @Override
+    @Test(expected = ResourceDoesNotSupportOperationException.class)
+    public void shouldGetAll() throws Exception {
+        super.shouldGetAll();
+    }
+
+    @Test
+    public void shouldUpdateObs() throws Exception {
+        ObsService obsService = Context.getObsService();
+        final String obsUuid ="47f18998-96cc-11e0-8d6b-9b9415a91465";
+        Date updatedObsDatetime = new Date();
+        double updatedValue = 10.0 ;
+
+        executeDataSet("obsDataSet2_2.xml");
+        Obs existingObs = obsService.getObsByUuid(obsUuid);
+        assertNotNull(existingObs);
+
+        String json = "{ \"value\":\"" + updatedValue + "\", \"person\":\"" + RestTestConstants1_8.PERSON_UUID
+                + "\", \"concept\":\"SNOMED CT:2332523\",\"obsDatetime\":\""+ DATE_FORMAT.format (updatedObsDatetime) +"\",\"voided\":false}";
+
+        handle(newPostRequest(getURI() + "/" + existingObs.getUuid(), json));
+        Obs updatedObs = obsService.getRevisionObs(existingObs);
+
+        assertNotNull(updatedObs);
+        assertEquals(10.0,updatedObs.getValueNumeric(),updatedValue);
+        assertEquals("da7f524f-27ce-4bb2-86d6-6d1d05312bd5",updatedObs.getPerson().getUuid());
```

Comment:
When making the POST, is the above person uuid the one you are passing?

Reply:
>"da7f524f-27ce-4bb2-86d6-6d1d05312bd5" =  RestTestConstants1_8.PERSON_UUID ...... who I passed in the json to be  updated

Reply:
Why a constant in one and hard coded value in the other?

Reply:
....Oh ..I get it ....